pkg/bpf/collection.go

```diff
@@ -464,15 +499,16 @@ func globalData(spec *ebpf.CollectionSpec) (varOffsets, error) {
 		case 1:
 			value = uint64(buf[0])
 		default:
-			return nil, fmt.Errorf("invalid variable size %d", vsi.Size)
+			return nil, nil, fmt.Errorf("invalid variable size %d", vsi.Size)
 		}
 
 		// Emit the variable's value by its offset in the datasec.
-		out[vsi.Offset] = value
+		out[vsi.Offset] = globalVar{v.Name, value}
+		names[v.Name] = struct{}{}
```

Comment:
We could use names as a two-way index:
```suggestion
		names[v.Name] = vsi.Offset
```
Then, in `inlineGlobalData`, we can use this to look up vars by name and merge overrides into values.